include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -662,6 +662,43 @@ def TT_DotOp : TT_Op<"dot", [Pure,
     let hasVerifier = 1;
 }
 
+
+//
+// DotScaled Op
+//
+def TT_DotScaledOp : TT_Op<"dot_scaled", [Pure,
+                             DotLike,
+                             TypesMatchWith<"result's type matches accumulator's type",
+                                            "d", "c", "$_self">]> {
+    let summary = "dot_scaled";
+
+    let description = [{
+        $d = matrix_multiply(scale($lhs, $lhs_scale), scale($rhs, $rhs_scale)) + $c.
+        Where scale(x, s) is a function that applies the scale per block following microscaling spec.
+    }];
+
+    let arguments = (
+      ins
+      // inputs are integer types as they are packed types and we currently
+      // don't have a representation for those.
+      TT_IntTensor:$lhs,
+      TT_IntTensor:$rhs,
+      TT_FloatTensor:$c,
+      TT_IntTensor:$lhs_scale,
+      Optional<TT_IntTensor>:$rhs_scale,
+      TT_F8F6F4TypeAttr:$lhs_type,
+      TT_F8F6F4TypeAttr:$rhs_type
+    );
+
+    let results = (outs TT_FloatTensor:$d);
+
+    // Not sure why I need to fully specify the optional group, but otherwise it complains when loading the mlir file
```

Comment:
what optional group?

Reply:
The one referring to the scale of the rhs being an std::optional

Reply:
I guess I still don't understand the meaning of the comment :) What does "I need to fully specify the optional group" means then?

Reply:
I meant that I had to specify all this funny syntax: ``(`,`) : (`,` $rhs_scale^ `,`)?``. In the other ops that we use optional args, we pretty much handle them in the frontend where we call different builders explicitly.
Here, I tried to handle the optional arg at the MLIR level, and it so happened that just writing ``(`,` $rhs_scale^)?`` wouldn't work. I had to write the whole if-else clause for the optional